sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala

```diff
@@ -271,4 +271,18 @@ private[spark] object SchemaUtils {
     }
     field._1
   }
+
+  /**
+   * @param str The string to be escaped.
+   * @return The escaped string.
+   */
+  def escapeMetaCharacters(str: String): String = {
```

Comment:
How about moving this into `org.apache.spark.sql.catalyst.util.StringUtils` instead?

Reply:
Hmm, I intended to use this for escaping strings in schema so I implemented this in `SchemaUtils`.
In fact, it's used for such purpose for now.
If we find it's useful for other purposes, how about moving this into a different place in the future?

Reply:
sgtm